### ISSUE TITLE
Invalid syntax when ternary elided by CFA

### ISSUE THREAD

**Issue by gavinpc-mindgrub** [rollup/rollup#3530]
- Rollup Version: 1.26
- Operating System (or Browser): Ubuntu Firefox 75
- Node Version (if applicable):
- Link to reproduction (IMPORTANT, read below):

Okay ([sandbox](https://rollupjs.org/repl/?version=2.7.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnNvbGUubG9nKCU3QiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1Q24lNUN0dGVzdCUzQSUyMHglMjAlM0QlM0UlNUNuJTVDdCU1Q3R4JTIwJTNEJTNEJTNEJTIwJTVDJTIyZm9vJTVDJTIyJTVDbiU1Q3QlNUN0JTVDdCUzRiUyMCU3QiU1Q24lNUN0JTVDdCU1Q3QlNUN0YXNzZXJ0JTNBJTIwJTVCJTVEJTJDJTVDbiU1Q3QlNUN0JTVDdCU3RCU1Q24lNUN0JTNBJTIwJTdCJTdEJTJDJTVDbiU3RCklNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyYW1kJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=))

Not okay: ([sandbox](https://rollupjs.org/repl/?version=2.7.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnNvbGUubG9nKCU3QiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1Q24lNUN0dGVzdCUzQSUyMHglMjAlM0QlM0UlNUNuJTVDdCU1Q3R0cnVlJTIwJTdDJTdDJTIweCUyMCUzRCUzRCUzRCUyMCU1QyUyMmZvbyU1QyUyMiU1Q24lNUN0JTVDdCU1Q3QlM0YlMjAlN0IlNUNuJTVDdCU1Q3QlNUN0JTVDdGFzc2VydCUzQSUyMCU1QiU1RCUyQyU1Q24lNUN0JTVDdCU1Q3QlN0QlNUNuJTVDdCUzQSUyMCU3QiU3RCUyQyU1Q24lN0QpJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmFtZCUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE))

### Expected Behavior

Bundle works the same as unbundled code.

### Actual Behavior

```
SyntaxError: expected expression, got '}'
```

The repro case is short, so I will provide it here for reference:

This works as expected:
```js
console.log({        
	test: x =>
		x === "foo"
			? {
				assert: [],
			}
	: {},
})
```

But if I add `true || ` to the condition, Rollup recognizes that it's always true and removes the ternary.  This is fine, except that it leaves the trailing comma (on a separate line), which is a syntax error:

Result with `true || `:

```js
define(function () { 'use strict';

	console.log({        
		test: x =>
			 {
					assert: [],
				}
		,
	});
});
```

Note that if I collapse the `assert` object onto a single line, the resulting code is also okay.

**Comment by lukastaegert** [comment]
I just learned that trailing commas cannot be placed on separate lines, oh my…

**Comment by lukastaegert** [comment]
Oh, actually, that is NOT the problem. It is more serious: The simplification moves the curly brace of the object directly after the `=>` arrow. Thus it is interpreted as a function body and we have a labeled statement followed by a `,`, which is not only illegal but also very much broken. On the positive side I can confirm from trying in several engines that you CAN have trailing commas on separate lines.

**Comment by lukastaegert** [comment]
So far there are at least two ways to trigger this:
https://rollupjs.org/repl/?version=2.7.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnNvbGUubG9nKHglMjAlM0QlM0UlMjB0cnVlJTIwJTI2JTI2JTIwJTdCeCUzQSUyMDQyJTJDJTIweSUzQSUyMDQzJTdEKSUzQiU1Q25jb25zb2xlLmxvZyh4JTIwJTNEJTNFJTIwdHJ1ZSUyMCUzRiUyMCU3QnglM0ElMjA0MiUyQyUyMHklM0ElMjA0MyU3RCUzQSUyMG51bGwpJTNCJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmFtZCUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by lukastaegert** [comment]
Fix at #3540

**Comment by gavinpc-mindgrub** [comment]
> it is interpreted as a function body and we have a labeled statement followed by a `,`

Ah, my fault for the false lead.  I did not quite dig into it fully.

Many thanks @lukastaegert for the quick turnaround!
